### PR TITLE
Handling of empty properties and nested properties

### PR DESCRIPTION
**Pull request by gdarmont**
Hi Spring Cloud Zookeeper team,

We are currently migrating projects from properties files to Spring Cloud Zookeeper (SCZ) and we encountered some problems with empty properties and nested properties.

For example, the following properties from a .properties file are not supported in the current version
```
foo.bar=value
foo.bar.baz=value2       --> not mapped in SCZ, since foo.bar is valued, children are not loaded
foo.bar2=                --> not mapped in SCZ, value is empty, we only search for children
```

Now, the following paths are supported :
```
/application/testProp=value        (previously supported)
/application/testProp/nested=value (not previously supported)
/application/testProp.dot=value    (previously supported)
/application/testProp.novalue=     (not previously supported)
```
Hope this helps,
Guillaume

### CODE REVIEW

**Comment by marcingrzejszczak** [comment]
Hi! Thanks for the contribution :) Before we go further - have you signed the Spring ICLA  form - https://support.springsource.com/spring_committer_signup ?

**Comment by gdarmont** [comment]
Hi Marcin, I've signed the ICLA for a previous (very small) PR. See #68.

File: spring-cloud-zookeeper-config/src/test/java/org/springframework/cloud/zookeeper/config/ZookeeperPropertySourceLocatorTests.java [hunk 1]
```diff
@@ -16,6 +16,11 @@
 
 package org.springframework.cloud.zookeeper.config;
 
+import static org.hamcrest.Matchers.equalTo;
```
Comment (marcingrzejszczak) [inline spring-cloud-zookeeper-config/src/test/java/org/springframework/cloud/zookeeper/config/ZookeeperPropertySourceLocatorTests.java]: Can you change the import in accordance with https://github.com/spring-cloud/spring-cloud-netflix/tree/master/eclipse ? If you're using Intellij Idea you can also import those there.

**Comment by marcingrzejszczak** [comment]
Except for the really minor issue looks good to me - @spencergibb ?

**Comment by gdarmont** [comment]
I've made the requested import changes. Sorry not having double checked this part.

**Comment by spencergibb** [comment]
:shipit:

**Pull request merged**